### ISSUE TITLE
View

### ISSUE THREAD

**Issue by trollfot** [pyrates/roll#75]
```
import inspect
from abc import ABC

class View(ABC):
    pass

def view_methods(cls):
   members = dict(inspect.getmembers(cls, predicate=inspect.isfunction))
   valid_methods = set(members.keys()) & set({'GET', 'POST'})
   for key in valid_methods:
       yield key, members[key]


class Roll:
    ...
    def view(self, path: str, **payload):
        def wrapper(cls):
            for meth, handler in view_methods(cls):
                    payload[meth] = handler
            self.routes.add(path, **payload)
            return cls
        return wrapper
```

**Comment by yohanboniface** [comment]
Is that a PR? :p

Do we need a new decorator? Can't we inspect the type of handler we receive?

**Comment by trollfot** [comment]
It's an idea i don't know how to submit to you.
We could or not, I don't know. This is how I use it currently.
I have also a way to pass an instance and get a partial with the self attached if it's a method and not a function. 
I don't know if it's useful to anyone, that's why I throw it here to discuss it.

**Comment by yohanboniface** [comment]
Yes, this is something I'd like to implement too.
There are situation where it's useful to deal with inheritance in views.
I'd more or less follow the Falcon pattern: `on_get`, `on_post` and so on.
Not sure though if we deal with class and/or instances (Falcon only accepts instances).

**Comment by yohanboniface** [comment]
Fixed by #80
